From 23bb9cab535219e873b04c45451e74efa1512f3b Mon Sep 17 00:00:00 2001
From: David Milburn <dmilburn@redhat.com>
Date: Mon, 18 Aug 2014 19:58:45 -0400
Subject: [block] nvme: Make iod bio timeout a parameter

Message-id: <1408391935-24886-13-git-send-email-dmilburn@redhat.com>
Patchwork-id: 87893
O-Subject: [RHEL7.1 PATCH BZ 1111259 12/22] NVMe: Make iod bio timeout a parameter
Bugzilla: 1111259
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Keith Busch <keith.busch@intel.com>

This was originally set to 4 times the IO timeout, but that was when
the IO timeout was 5 seconds instead of 30. 20 seconds for total time
to failure seemed more reasonable than 2 minutes for most, but other
users have requested to make this a module parameter instead.

Signed-off-by: Keith Busch <keith.busch@intel.com>
[renamed the module parameter to retry_time]
[made retry_time static]
Signed-off-by: Matthew Wilcox <matthew.r.wilcox@intel.com>

(cherry picked from commit 61e4ce086df0a64a555880089e3b782517c828c0)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/block/nvme-core.c b/drivers/block/nvme-core.c
index 4f0f4b1..e16682d 100644
--- a/drivers/block/nvme-core.c
+++ b/drivers/block/nvme-core.c
@@ -48,12 +48,16 @@
 #define SQ_SIZE(depth)  (depth * sizeof(struct nvme_command))
 #define CQ_SIZE(depth)  (depth * sizeof(struct nvme_completion))
 #define ADMIN_TIMEOUT (60 * HZ)
-#define IOD_TIMEOUT (4 * NVME_IO_TIMEOUT)
+#define IOD_TIMEOUT (retry_time * HZ)
 
 unsigned char io_timeout = 30;
 module_param(io_timeout, byte, 0644);
 MODULE_PARM_DESC(io_timeout, "timeout in seconds for I/O");
 
+static unsigned char retry_time = 30;
+module_param(retry_time, byte, 0644);
+MODULE_PARM_DESC(retry_time, "time in seconds to retry failed I/O");
+
 static int nvme_major;
 module_param(nvme_major, int, 0);
 
-- 
1.7.1